docs/DeveloperGuide.md

```diff
@@ -1,34 +1,851 @@
+    * Under the `Project SDK:` section, ensure that the correct SDK (**java version "11.0.8"** or higher) is selected.
+1. Import the project as a Gradle project.
+    * At the same welcome page, click on `Open or Import`.
+    * Navigate to the folder where you cloned the repo, locate the `build.gradle` file within the folder and select it.
+    * Choose the `Open as Project` option when asked.
+    * The project may take a couple of minutes to import depending on your actual computer hardware setup.
+1. Verify the setup.
+    * Under the `seedu.modtracker` package, locate the `ModTracker` class and run it.
+    * Try a few commands. You may want to refer to the [user guide](https://ay2021s1-cs2113t-f12-4.github.io/tp/UserGuide.html).
+    * Run the [tests]() to ensure they all pass.
+
+### 2.2 Before writing the code <a name="2.2"></a>
+1. Configure the coding style
+    * If using IntelliJ IDEA, follow this [guide](https://se-education.org/guides/tutorials/intellijCodeStyle.html) to set up IDEA's coding style yo match ours.
+    
+1. Set up CI
+    * This project comes with a GitHub Actions config files (in `.github/workflows` folder). When GitHub detects those files, 
+    it will run the CI for the project automatically at each push to the `master` branch or to any PR.
+    No setup is required for this.
+    
+1. Learn the design
+    * When you are ready to start coding, we recommend that you look at [ModTracker's architecture](#architecture) to get some 
+    sense of the overall design.
+
+
+## 3. Design <a name="3"></a>
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### 3.1 Architecture <a name="3.1"></a>
+![Architecture](diagrams/Architecture.png)
+
+`Main` has a `ModTracker` class. It is the main entry point for the ModTracker application. It contains the main, loadData, run, 
+startAndGetName and runCommandLoopUntilExitCommand methods. The methods are responsible for:
+* At app launch:
+    * Creation of a ModTracker object.
+    * Load previously saved data from the text file into ModTracker.
+    * Run the ModTracker app.
+* At runtime:
+    * Continuously prompt the user for input until app termination.
+* At shut down:
+    * Save user data into a text file.
+    * Invoke clean up methods where necessary.
+    
+The rest of the app consists of 4 packages:
+* `Ui` : The user interface of the app.
+* `Parser` : Parses the user input and calls the corresponding methods.
+* `Model` : Holds the data of the app in-memory, containing the classes: ModuleList, Module, TaskList and Task.
+* `Storage` : Saves and loads data to and from the hard disk.
+
+##### How the architecture components interact with each other
+
+The Sequence Diagram below shows how the components interact with each other for the scenario where the user types 
+the command `addmod CS2113T`.  
+
+![highlevelsequencediagram](diagrams/highlevelsequencediagram.png)
```

Comment:
![image](https://user-images.githubusercontent.com/60372589/97387177-7818e280-1910-11eb-84f2-b2bae8bc04b4.png)

I think it is unnecessary to use quotation mark, maybe you can just call the variable